### PR TITLE
Add CampaignAutomationImplementingTests

### PR DESCRIPTION
**Pull request by rossrossp**

### CODE REVIEW

File: CampaignAutomation/Tests/CampaignAutomationImplementingTests.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+import XCTest
+@testable import RealifeTech
+
+class CampaignAutomationImplementingTests: XCTestCase {
+
+    private var mockAnalytics: MockAnalytics!
+    private var sut: CampaignAutomationImplementing!
+
+    override func setUpWithError() throws {
```
Comment (You-Hsuan) [inline CampaignAutomation/Tests/CampaignAutomationImplementingTests.swift:17]: Could we call `try super.setUpWithError()` here please? Thank you

Reason being that Apple doesn't guarantee there won't be changes in the superclass, so it's best to keep calling super. :) (Discussion: https://developer.apple.com/forums/thread/113314)

File: CampaignAutomation/Tests/CampaignAutomationImplementingTests.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        mockAnalytics = MockAnalytics()
+        sut = CampaignAutomationImplementing(defaultFetcher: RLTViewFetcher(),
+                                             analyticsLogger: mockAnalytics)
+    }
+
+    override func tearDownWithError() throws {
+        mockAnalytics = nil
+        sut = nil
```
Comment (You-Hsuan) [inline CampaignAutomation/Tests/CampaignAutomationImplementingTests.swift:25]: Could we call `try super. tearDownWithError() ` please? Thank you.
The same reason as above.

File: CampaignAutomation/Tests/CampaignAutomationImplementingTests.swift (hunk 1 above)
Comment (rossrossp) [reply CampaignAutomation/Tests/CampaignAutomationImplementingTests.swift:17]: Done.

File: CampaignAutomation/Tests/CampaignAutomationImplementingTests.swift (hunk 2 above)
Comment (rossrossp) [reply CampaignAutomation/Tests/CampaignAutomationImplementingTests.swift:25]: Done.

**Review by You-Hsuan — APPROVED** [review]

**Pull request merged**